Marlin/src/lcd/menu/menu_main.cpp

```diff
@@ -43,7 +43,12 @@ void lcd_pause() {
   #endif
 
   #if ENABLED(PARK_HEAD_ON_PAUSE)
-    pause_print(PAUSE_PARK_RETRACT_LENGTH, NOZZLE_PARK_POINT, 0, true);
+    lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_INIT, ADVANCED_PAUSE_MODE_PAUSE_PRINT, active_extruder);
+    if (pause_print(PAUSE_PARK_RETRACT_LENGTH, NOZZLE_PARK_POINT, 0, true)) {
```

Comment:
It would be better to avoid doing this here, since this point now becomes the main loop. It would be better if a G-code were called from here instead. If `M25` and `M125` are unsuitable, either extend them with "undocumented" parameters that invoke the behavior here, or add a new (undocumented) G-code that invokes this behavior.